Fixes Cli adds tests for linux

fundies opened the pull request:
@JoshDreamland  needs to enable travis-ci for this repo

https://travis-ci.org/fundies/enigma-dev <- tests run under mine

JoshDreamland commented on Makefile:
```diff
@@ -13,5 +13,8 @@ Game:
 	$(MAKE) -C ENIGMAsystem/SHELL clean
+	
+emake:
+	$(MAKE) -j 3 -C CommandLine/emake/
```
It's a bit tacky to assign a thread count yourself, but whatever.

RobertBColton commented on CommandLine/emake/EnigmaCallbacks.cpp:
```diff
@@ -0,0 +1,92 @@
+
+void* CallBack::OutputThread(void*)
+{
+    if (_outFile)
```
Two space is my default now in Notepad++ for C++ ok? kthx I use tabs in my own projects though. :smile:

fundies replied:
There's tons of UpperCamel in the project here's one example of many https://github.com/enigma-dev/enigma-dev/blob/d9746dfc6d557c4c22c8bd8cf24e02e3272eade0/ENIGMAsystem/SHELL/Universal_System/bufferstruct.h#L94

`Game maker uses lower_func and this project mostly uses  a fair mix of everything. You really need to post style guidelines somewhere because I have no idea what you want me to use?

class OptionsParser or class options_parser or class optionsParser?

void DoThis() or void do_this() or void doThis()?

samething for enums and arguments... I really can't tell what you want

RobertBColton replied:
Only the compiler stuff should be camel case, snake case is for GML functions only. I think that's pretty obvious.

You should generally stick to two spaces for indent and only worry about changing files that are a super messy mixture of tabs and spaces. If some file is all tabs, just leave it be.

JoshDreamland replied:
The problem is that some of the compiler stuff is using UpperCamel rather than lowerCamel for functions. Only class names should begin with a capital letter. Capitalizing function names is a C# standard that I don't agree with.

RobertBColton replied:
How about Camel Toe?

fundies commented:
I fixed the spaces added quotes to the clean command and removed the "Error" and ""-j3". I need further input on how I should name my classes and functions though.

JoshDreamland commented on CommandLine/emake/EnigmaPlugin.hpp:
```diff
@@ -0,0 +1,55 @@
+{
+  PLUGIN_ERROR = 1,
+  PLUGIN_SUCCESS = 0
+} PluginResult;
```
Why C-style enums?

JoshDreamland commented on CommandLine/emake/OptionsParser.cpp:
```diff
@@ -0,0 +1,505 @@
+#include "OptionsParser.hpp"
```
This entire file can be replaced with a few lines of code atop [GFlags](https://gflags.github.io/gflags/), or my own [DeepFlags](https://github.com/JoshDreamland/DeepFlags/blob/master/DeepFlags.hpp), though I think the latter is overkill.

JoshDreamland commented on ENIGMAsystem/SHELL/Universal_System/Extensions/MotionPlanning/motion_planning_struct.h:
```diff
@@ -35,6 +35,14 @@ using std::multimap;
 #endif
 
+#ifdef COLLISION_NONE
+	#error The motion planning extension requires a collision system.
```
It would be nice if new code didn't use tabs, but I suppose this isn't going to kill anyone.
